sam_view.c

```diff
@@ -291,6 +293,8 @@ int main_samview(int argc, char *argv[])
                 settings.subsam_seed = rand();
             }
             settings.subsam_frac = strtod(q, &q);
+            if (settings.subsam_frac < 1e-10)
+                print_error_errno("view", "Could not parse sampling argument \"%s\"", optarg);
```

Comment:
The best way to see if `strtod` succeeded is to look at `q` and see if it consumed any characters.